packages/phoenix-event-display/src/tests/loaders/phoenix-loader.test.ts

```diff
@@ -0,0 +1,112 @@
+/**
+ * @jest-environment jsdom
+ */
+import { InfoLogger } from '../../helpers/info-logger';
+import { PhoenixLoader } from '../../loaders/phoenix-loader';
+import { ThreeManager } from '../../managers/three-manager/index';
+import { UIManager } from '../../managers/ui-manager/index';
+
+jest.mock('../../managers/three-manager/index');
+
+describe('PhoenixLoader', () => {
+  let phoenixLoader: PhoenixLoader;
+  const eventData = {
+    Event: {
+      'event number': 1,
+      'run number': 1,
+      Hits: {
+        hitsCollection: [
+          {
+            pos: [
+              -2545.135009765625, -2425.1064453125, 7826.09912109375,
+              -2545.135009765625, -1.1222461462020874, 7826.09912109375,
+            ],
+            type: 'Line',
+          },
+        ],
+      },
+    },
+    Event2: {
+      'event number': 2,
+      'run number': 2,
+      Hits: {
+        hitsCollection: [
+          {
+            pos: [
+              -2545.135009765625, -2425.1064453125, 7826.09912109375,
+              -2545.135009765625, -1.1222461462020874, 7826.09912109375,
+            ],
+            type: 'Line',
+          },
+        ],
+      },
+    },
+  };
+
+  beforeEach(() => {
+    phoenixLoader = new PhoenixLoader();
+
+    const graphicsLibrary = new ThreeManager(new InfoLogger());
+    const uiManager = new UIManager(new ThreeManager(new InfoLogger()));
+    const infoLogger = new InfoLogger();
+
+    phoenixLoader.buildEventData(
+      eventData,
+      graphicsLibrary,
+      uiManager,
+      infoLogger
+    );
+  });
+
+  it('should create an instance', () => {
+    expect(phoenixLoader).toBeTruthy();
+  });
+
+  it('should get the list of event names from the event data', () => {
+    const eventsList = phoenixLoader.getEventsList(eventData);
+
+    expect(eventsList).toHaveLength(2);
+    expect(eventsList[0]).toBe('Event');
+    expect(eventsList[1]).toBe('Event2');
+  });
+
+  it('should get list of collections in the event data', () => {
+    const collectionList = phoenixLoader.getCollections();
+
+    expect(collectionList).toEqual([
+      'event number',
+      'run number',
+      'Hits',
+      'event number',
+      'run number',
+      'Hits',
+    ]);
+  });
+
+  it('should get the collection with the given collection name from the event data', () => {
+    const collectionName = eventData['Event']['Hits'];
+
+    const collection = phoenixLoader.getCollection('Hits');
+
+    expect(collection).toEqual(collectionName);
+  });
+
+  it('get metadata associated to any event', () => {
+    // as of now returns an empty array of objects due to the event data we add in this test suite
+    const metadata = phoenixLoader.getEventMetadata();
+
+    expect(metadata).toEqual([]);
```

Comment:
I get it now. It's empty because `buildData` accepts a single event not the file which contains multiple events.

So I recommend you pass `eventData['Event']` to `phoenixLoader.buildEventData` for this to work.

Reply:
Or you can pass in an empty object `threeManager.init({})`.

Reply:
If we use the real `ThreeManager` and initialize it, we know it calls the constructor for the `RendererManager` and this line throws up the error which we have been seeing since the start -

https://github.com/HSF/phoenix/blob/50d629e7dbf4432e3f1dd8b9a49466b53ecc0fc8/packages/phoenix-event-display/src/managers/three-manager/renderer-manager.ts#L20

```
Error creating WebGL context.
```

In these types of situations, we have to use a mocked ThreeManager (+ its functions) every time we need it in our tests.

Reply:
The mock is fine. Looks okay now. But I think one last improvement could be to use the same object but just mock one method.

```ts
beforeEach(() => {
  phoenixLoader = new PhoenixLoader();

  const infoLogger = new InfoLogger();
  const threeManager = new ThreeManager(infoLogger);

  jest.spyOn(threeManager, 'addEventDataTypeGroup').mockReturnValue(new Group());

  phoenixLoader.buildEventData(
    eventData['Event'],
    threeManager,
    uiManager,
    infoLogger
  );
});
```

Reply:
cool!